client/my-sites/sidebar/sidebar.jsx

```diff
@@ -415,7 +415,7 @@ export class MySitesSidebar extends Component {
 			return null;
 		}
 
-		if ( ! config.isEnabled( 'manage/people' ) && site.options ) {
+		if ( site.options ) {
```

Comment:
This redirects all users to `/wp-admin/users.php` to manage users - the whole block should be removed.